proposed/http-message.md

```diff
@@ -776,18 +776,32 @@ interface ServerRequestInterface extends RequestInterface
     public function withQueryParams(array $query);
 
     /**
-     * Retrieve the upload file metadata.
+     * Retrieve the uploaded files.
      *
-     * This method MUST return file upload metadata in the same structure
+     * This method MUST return the file upload metadata in the same structure
      * as PHP's $_FILES superglobal.
      *
      * These values MUST remain immutable over the course of the incoming
      * request. They SHOULD be injected during instantiation, such as from PHP's
      * $_FILES superglobal, but MAY be derived from other sources.
      *
-     * @return array Upload file(s) metadata, if any.
+     * @return UploadedFileInterface[] The uploaded file(s), if any.
```

Comment:
> The same is actually true for getQueryParams() and getParsedBody().

Except that these are *always* unstructured, which is a very different situation. `$_FILES` has a well-documented (if sometimes vexing!) structure.

In this case, I think we can only have a return typehint of `array`, and would have to indicate that it returns either an array of `UploadedFileInterface` instances, or a nested tree of them, as a description for the return value annotation.